### PR TITLE
ci: add workflow for releasing build artifacts

### PR DESCRIPTION
**Pull request by true1ann**
Workflow:
1. Install node 23
2. Install Bun
3. Install dependencies
4. Build (minified)
5. Create a release
6. Push both artifacts as release assets

### CODE REVIEW

**Review by Ushie — CHANGES_REQUESTED** [review]
What's the point of bun if you end up using node and npm?

**Comment by true1ann** [comment]
Ive changed the workflow and now it does not install node&npm, just bun

**Review by nexpid — CHANGES_REQUESTED** [review]
Use https://github.com/oven-sh/setup-bun

**Comment by true1ann** [comment]
okay, now using `oven-sh/setup-bun` over manual install

**Review by nexpid — APPROVED** [review]
LGTM

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Build & Release
```
Comment (Ushie) [inline .github/workflows/main.yml]: This implies releasing to GitHub releases, which it doesn't, a more fitting name is needed

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+on:
+  push:
+    branches:
+      - main
+      - release-branch
+  pull_request:
+    branches:
+      - main
+      - release-branch
```
Comment (Ushie) [inline .github/workflows/main.yml]: What is release-branch?

**Review by Ushie — CHANGES_REQUESTED** [review]

**Review by Ushie — CHANGES_REQUESTED** [review]
File name should be updated to something more fitting

**Comment by true1ann** [comment]
`build-artifact.yml`?

**Comment by Ushie** [comment]
> `build-artifact.yml`?

Sounds good to me, also fits for the workflow name

**Review by Ushie — APPROVED** [review]

**Pull request merged**